backend/song/api/views.py

```diff
@@ -0,0 +1,121 @@
+from rest_framework import generics
+
+from backend.song import models
+from backend.song.permissions import IsOwnerOrReadOnly
+from backend.song.api import serializers
+
+
+class CategoriesList(generics.ListCreateAPIView):
```

Comment:
The idea was to have a view for `POST` and `GET` requests and another one for `PUT/PATCH` and `DELETE` so that we can have a clear distinction in URL specifications

Reply:
can we combine all in the same endpoint?

Reply:
django rest_framework does not provide a combination of all the 4 HTTP methods under one class it's probably for a reason but we can still customize it if you really want to have them under one class